contrib/pkg/clusterpool/clusterpool.go

```diff
@@ -0,0 +1,365 @@
+package clusterpool
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/homedir"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	awsutils "github.com/openshift/hive/contrib/pkg/utils/aws"
+	azureutils "github.com/openshift/hive/contrib/pkg/utils/azure"
+	gcputils "github.com/openshift/hive/contrib/pkg/utils/gcp"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/clusterresource"
+)
+
+const (
+	cloudAWS   = "aws"
+	cloudAzure = "azure"
+	cloudGCP   = "gcp"
+
+	longDesc = `
+OVERVIEW
+The hiveutil clusterpool create-pool command generates and applies the
+artifacts needed to create a new Hive cluster pool.
+Currently cluster pools are supported in AWS, Azure and GCP.
+
+IMAGES
+An existing ClusterImageSet can be specified with the --image-set
+flag. Otherwise, one will be generated using the image specified for the
+cluster pool using --release-image flag. If neither of these flags are 
+specified, default image is obtained from the following URL:
+https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/4-stable/latest
+
+ENVIRONMENT VARIABLES
+The command will use the following environment variables for its output:
+
+PULL_SECRET - If present, it is used as the cluster pool's pull
+secret and will override the --pull-secret flag. If not present, and
+the --pull-secret flag is not specified, then the --pull-secret-file is
+used.
```

Comment:
There is still no way for the user to determine what the name of the default pull secret file is if it does not already exist.